### PR TITLE
[28기 이가윤] 메인 페이지 구현완료

### PR DESCRIPTION
**Pull request by Gayun00**
## :: 최근 작업 주제 (하나 이상의 주제를 선택해주세요.)
- [x] 기능 추가
- [ ] 리뷰 반영
- [ ] 리팩토링
- [ ] 버그 수정
- [ ] 컨벤션 수정

<br />

[Add] subSlider 레이아웃  구현
 - 목업데이터 입력
 - css 진행중

[Modify] mainslider 수정

[Modify] subslider 수정

[Modify] 너비 고정

[Modify] css 수정 중

## :: 구현 목표 (해당 브랜치에서 구현하고자 하는 하나의 목표를 설정합니다.)
- 아이디, 비밀번호 입력창을 사용한 로그인 기능 구현 (예시)

<br />

## :: 구현 사항 설명 (작업한 내용을 상세하게 기록합니다.)

<br />

## :: 성장 포인트 (해당 기능을 구현하며 고민했던 사항이나 새로 알게된 부분, 어려웠던 점 등을 작성합니다.)

<br />

## :: 기타 질문 및 특이 사항
subSliderItem 컴포넌트가 화면에 표시되지 않습니다.
package.json, package-lock.json파일 깃 충돌 수정 후 일어난 오류라 뭔가 잘못 건드린 것 같습니다.
props가 잘 전달되었는지는 확인했는데 slick라이브러리에 오류가 생겨 표시가 되지 않는 것 같습니다. 어떻게 해야할지 잘 모르겠습니다.

### CODE REVIEW

File: src/pages/Main/Main.js [hunk 1]
```diff
@@ -1,7 +1,28 @@
-import React from 'react';
+import React, { Component, useEffect, useState } from 'react';
```
Comment (ghost) [inline src/pages/Main/Main.js]: __[ESLint]__ 'Component' is defined but never used. ([view](https://sider.review/gh/repos/446331007/pulls/6#issue-6480806))

<details>

| Rule | Severity | Recommended |
| ---- | -------- | ----------- |
| `no-unused-vars` | `warn` | Yes |

References:
- <https://eslint.org/docs/rules/no-unused-vars>

You can [close this issue](https://sider.review/gh/repos/446331007/pulls/6#issue-6480806) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).

</details>

File: src/pages/Main/Main.js [hunk 2]
```diff
@@ -1,7 +1,28 @@
...
+import Slider from 'react-slick';
```
Comment (ghost) [inline src/pages/Main/Main.js]: __[ESLint]__ 'Slider' is defined but never used. ([view](https://sider.review/gh/repos/446331007/pulls/6#issue-6480807))

<details>

| Rule | Severity | Recommended |
| ---- | -------- | ----------- |
| `no-unused-vars` | `warn` | Yes |

References:
- <https://eslint.org/docs/rules/no-unused-vars>

You can [close this issue](https://sider.review/gh/repos/446331007/pulls/6#issue-6480807) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).

</details>

File: src/pages/Main/MainSlider.js [hunk 3]
```diff
@@ -0,0 +1,148 @@
+import React, { Component, useRef } from 'react';
```
Comment (ghost) [inline src/pages/Main/MainSlider.js]: __[ESLint]__ 'Component' is defined but never used. ([view](https://sider.review/gh/repos/446331007/pulls/6#issue-6480808))

<details>

| Rule | Severity | Recommended |
| ---- | -------- | ----------- |
| `no-unused-vars` | `warn` | Yes |

References:
- <https://eslint.org/docs/rules/no-unused-vars>

You can [close this issue](https://sider.review/gh/repos/446331007/pulls/6#issue-6480808) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).

</details>

File: src/pages/Main/SubSlider/SubSlider.js [hunk 4]
```diff
@@ -0,0 +1,146 @@
+import React, { Component } from 'react';
```
Comment (ghost) [inline src/pages/Main/SubSlider/SubSlider.js]: __[ESLint]__ 'Component' is defined but never used. ([view](https://sider.review/gh/repos/446331007/pulls/6#issue-6480809))

<details>

| Rule | Severity | Recommended |
| ---- | -------- | ----------- |
| `no-unused-vars` | `warn` | Yes |

References:
- <https://eslint.org/docs/rules/no-unused-vars>

You can [close this issue](https://sider.review/gh/repos/446331007/pulls/6#issue-6480809) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).

</details>

File: src/pages/Main/SubSlider/SubSlider.js [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+    z-index: 1;
+  }
+`;
+
+const ButtonWrapper = styled.div`
+  position: absolute;
+  top: 69%;
+`;
+
+const ImgContainer = styled.div`
```
Comment (ghost) [inline src/pages/Main/SubSlider/SubSlider.js:103]: __[ESLint]__ 'ImgContainer' is assigned a value but never used. ([view](https://sider.review/gh/repos/446331007/pulls/6#issue-6480810))

<details>

| Rule | Severity | Recommended |
| ---- | -------- | ----------- |
| `no-unused-vars` | `warn` | Yes |

References:
- <https://eslint.org/docs/rules/no-unused-vars>

You can [close this issue](https://sider.review/gh/repos/446331007/pulls/6#issue-6480810) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).

</details>

File: src/pages/Main/MainSlider.js [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+import 'slick-carousel/slick/slick-theme.css';
+import { IoIosArrowForward, IoIosArrowBack } from 'react-icons/io';
+
+function MainSlider({ mainSlide }) {
+  const prevButtonRef = useRef();
+  const nextButtonRef = useRef();
+
+  function ArrowLeft(props) {
+    return (
+      <button {...props} className={'prev'}>
```
Comment (ghost) [inline src/pages/Main/MainSlider.js]: __[ESLint]__ Curly braces are unnecessary here. ([view](https://sider.review/gh/repos/446331007/pulls/6#issue-6480811))

<details>

| Rule | Severity | Recommended |
| ---- | -------- | ----------- |
| `react/jsx-curly-brace-presence` | `warn` | No |

References:
- <https://github.com/yannickcr/eslint-plugin-react/tree/master/docs/rules/jsx-curly-brace-presence.md>

You can [close this issue](https://sider.review/gh/repos/446331007/pulls/6#issue-6480811) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).

</details>

File: src/pages/Main/SubSlider/SubSlider.js [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+import styled from 'styled-components';
+import 'slick-carousel/slick/slick.css';
+import 'slick-carousel/slick/slick-theme.css';
+import { IoIosArrowForward, IoIosArrowBack } from 'react-icons/io';
+import SubSliderItem from './SubSliderItem';
+
+function MainSlider({ mainSlide }) {
+  function ArrowLeft(props) {
+    return (
+      <button {...props} className={'prev'}>
```
Comment (ghost) [inline src/pages/Main/SubSlider/SubSlider.js]: __[ESLint]__ Curly braces are unnecessary here. ([view](https://sider.review/gh/repos/446331007/pulls/6#issue-6480813))

<details>

| Rule | Severity | Recommended |
| ---- | -------- | ----------- |
| `react/jsx-curly-brace-presence` | `warn` | No |

References:
- <https://github.com/yannickcr/eslint-plugin-react/tree/master/docs/rules/jsx-curly-brace-presence.md>

You can [close this issue](https://sider.review/gh/repos/446331007/pulls/6#issue-6480813) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).

</details>

File: src/pages/Main/SubSlider/SubSlider.js [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+        <IoIosArrowBack className="arrowBack" />
+      </button>
+    );
+  }
+  function ArrowRight(props) {
+    return (
+      <button {...props} className={'next'}>
```
Comment (ghost) [inline src/pages/Main/SubSlider/SubSlider.js]: __[ESLint]__ Curly braces are unnecessary here. ([view](https://sider.review/gh/repos/446331007/pulls/6#issue-6480814))

<details>

| Rule | Severity | Recommended |
| ---- | -------- | ----------- |
| `react/jsx-curly-brace-presence` | `warn` | No |

References:
- <https://github.com/yannickcr/eslint-plugin-react/tree/master/docs/rules/jsx-curly-brace-presence.md>

You can [close this issue](https://sider.review/gh/repos/446331007/pulls/6#issue-6480814) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).

</details>

File: src/pages/Main/SubSlider/SubSlider.js [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+    prevArrow: <ArrowLeft />,
+    nextArrow: <ArrowRight />,
+  };
+
+  return (
+    <Wrapper>
+      <h1>오직 스테이윗미에서만</h1>
+      <Slider {...settings}>
+        {mainSlide.map(slide => (
+          <SubSliderItem />
```
Comment (ghost) [inline src/pages/Main/SubSlider/SubSlider.js]: __[ESLint]__ Missing "key" prop for element in iterator ([view](https://sider.review/gh/repos/446331007/pulls/6#issue-6480816))

<details>

| Rule | Severity | Recommended |
| ---- | -------- | ----------- |
| `react/jsx-key` | `warn` | Yes |

References:
- <https://github.com/yannickcr/eslint-plugin-react/tree/master/docs/rules/jsx-key.md>

You can [close this issue](https://sider.review/gh/repos/446331007/pulls/6#issue-6480816) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).

</details>

File: src/pages/Main/MainSlider.js [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+          <ImgContainer>
+            <img src={slide.img} alt="" />
+          </ImgContainer>
+        ))}
+      </Slider>
+      <ButtonWrapper>
+        <div className="slideNum">
+          <p className="currentSlideNum">1 / 10</p>
+        </div>
+        <div className="slideButtons"></div>
```
Comment (ghost) [inline src/pages/Main/MainSlider.js]: __[ESLint]__ Empty components are self-closing ([view](https://sider.review/gh/repos/446331007/pulls/6#issue-6480817))

<details>

| Rule | Severity | Recommended |
| ---- | -------- | ----------- |
| `react/self-closing-comp` | `warn` | No |

References:
- <https://github.com/yannickcr/eslint-plugin-react/tree/master/docs/rules/self-closing-comp.md>

You can [close this issue](https://sider.review/gh/repos/446331007/pulls/6#issue-6480817) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).

</details>

File: public/data/subslider.json [hunk 11]
```diff
@@ -0,0 +1,37 @@
+[
+  {
+    "title": "오직 스테이윗미에서만",
+    "hotels": [
+      {
+        "img": "",
+        "name": "베로마",
+        "location": "부산/해운대구",
+        "price": "2000~25000"
```
Comment (bonbon0202) [inline public/data/subslider.json]: price range 이죠?! 아마 백엔드에서는 가격을 string이 아닌 minPrice, maxPrice 등의 key 값으로 Int (number)  타입으로 관리할 것 같은데 확인해주세요!

File: src/pages/Main/Main.js [hunk 12]
```diff
@@ -1,7 +1,27 @@
...
 
 function Main() {
-  return <div>main</div>;
+  const [mainSlide, setMainSlide] = useState([]);
+  useEffect(() => {
+    fetch('/data/mainSlider.json')
+      .then(res => res.json())
+      .then(res => setMainSlide([...res]));
+  }, []);
+  return (
```
Comment (bonbon0202) [inline src/pages/Main/Main.js]: 가독성을 위해 useEffect 기준으로 위 아래, 한 줄씩만 띄어주세요-!

File: src/pages/Main/MainSlider.js [hunk 13]
```diff
@@ -0,0 +1,146 @@
...
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  width: 100%;
+  height: 100%;
+  background-color: orange;
+
+  img {
+    object-fit: cover;
+    /* margin: 0 5rem 0 5rem; */
```
Comment (bonbon0202) [inline src/pages/Main/MainSlider.js]: 불필요한 주석이면 삭제해주세요!

File: src/pages/Main/MainSlider.js [hunk 14]
```diff
@@ -0,0 +1,146 @@
...
+`;
+
+const SliderWrapper = styled.div`
+  height: 500px;
+  overflow: hidden;
+`;
+
+const ButtonWrapper = styled.div`
+  position: absolute;
+  top: 69%;
```
Comment (bonbon0202) [inline src/pages/Main/MainSlider.js]: top에서 69%이면 bottom을 기준으로 하는 것은 어떨까요?! 디스플레이의 크기에 따라 오차범위가 클 것 같네요!

File: src/pages/Main/SubSlider/SubSliderItem.js [hunk 15]
```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import styled from 'styled-components';
+
+function SubSliderItem() {
+  return (
+    <Wrapper>
```
Comment (bonbon0202) [inline src/pages/Main/SubSlider/SubSliderItem.js]: - styled-component를 이용해서 스타일링이 된 태그들은 해당 definition으로 이동해서 바로 스타일링을 확인이 가능한데 className으로 선언한 부분은 바로 definition으로 이동이 불가능해서 스타일링 파악하기에 어려움이 있습니다.
- 위의 내용에 덧붙여서 className을 이용한 스타일링의 nesting depth가 깊어지는 경우에는 더 더욱 찾기가 힘들어집니다
- depth가 깊어지는 경우 뿐만 아니라 1depth로도 className을 이용한 스타일 선언부가 길어지는 경우에도 해당 스타일링을 찾기 어려워집니다.
- 위의 사항들로 인해서 CSS in JS(styled-components 등)을 이용한 스타일링을 하는 경우에는 라이브러리 사용상 강제되는 등의 불가피한 부분을 제외하고는 className을 통한 스타일링은 최대한 지양하는게 권장됩니다.

**Review by bonbon0202 — COMMENTED** [review]
가윤님! slick 이용하셔서 슬라이드 구현해주셨군요! 
잘 해주셨는데 중간중간 참조하실 부분에 대한 코멘트 드렸습니다! 일단, 불필요한 주석처리도 모두 지워주시고 다시 한 번 더 push 해주세요!

File: src/pages/Main/MainSlider.js [hunk 16]
```diff
@@ -5,79 +5,69 @@ import 'slick-carousel/slick/slick.css';
...
+  //     </button>
+  //   );
+  // }
+  // function ArrowRight(props) {
+  //   return (
+  //     <button {...props} className="next">
+  //       <IoIosArrowForward className="arrowForward" />
+  //     </button>
+  //   );
+  // }
```
Comment (bonbon0202) [inline src/pages/Main/MainSlider.js]: 불필요하신 주석이라면 모두 삭제해주세요!

File: src/pages/Main/SubSlider/SubSlider.js [hunk 17]
```diff
@@ -1,77 +1,98 @@
...
-        ))}
-      </Slider>
-      <div className="slideNum">
-        <p className="currentSlideNum">1 / 10</p>
-      </div>
-      <div className="slideButtons" />
+      <SliderTitle>{subSlide[0].title}</SliderTitle>
+      <SliderWrapper>
+        <Slider {...settings}>
+          {subSlide[0].hotels.map((hotel, idx) => (
```
Comment (bonbon0202) [inline src/pages/Main/SubSlider/SubSlider.js:21]: 보통은 데이터에 접근할 때, 인덱스로 접근하는 경우는 불안정하기 때문에 이런 경우는 객체 안에 배열로 있는 데이터 구조가 더 적합할 것 같네요-! 참조만해주세요.

File: src/pages/Main/SubSlider/SubSlider.js [hunk 18]
```diff
@@ -1,77 +1,98 @@
...
+  width: 80vw;
+`;
+
+const SliderTitle = styled.h2`
+  margin-bottom: 0.5rem;
+  text-align: center;
+  font-size: 1.5rem;
+`;
+
+const SliderWrapper = styled.div``;
```
Comment (bonbon0202) [inline src/pages/Main/SubSlider/SubSlider.js:101]: style이 불필요한 요소는 컴포넌트가 아닌 그냥 '요소'로 적어주셔도 됩니다.

**Review by bonbon0202 — COMMENTED** [review]
가윤님! 불필요한 주석이 너무 많이 보이는 것 같습니다. 
주석만 삭제해주시고 다른 리뷰는 참조해주세요.

File: src/pages/Main/SubSlider/SubSlider.js (hunk 18 above)
Comment (Gayun00) [reply src/pages/Main/SubSlider/SubSlider.js:101]: 넵 알겠습니다!

File: src/pages/Main/SubSlider/SubSlider.js (hunk 17 above)
Comment (Gayun00) [reply src/pages/Main/SubSlider/SubSlider.js:21]: 넵!

**Pull request merged**